vault/hcp_link/proto/meta/meta.proto

```diff
@@ -34,6 +34,57 @@ message ListAuthResponse {
   repeated Auth Auths = 1;
 }
 
+message GetClusterStatusRequest {}
+
+message HANode {
+  string Hostname = 1;
+}
+
+message RaftServer {
+  // NodeID is the name of the server
+  string NodeID = 1;
+
+  // Address is the IP:port of the server, used for Raft communications
+  string Address = 2;
+
+  // Leader is true if this server is the current cluster leader
+  bool Leader = 3;
+
+  // Protocol version is the raft protocol version used by the server
+  string ProtocolVersion = 4;
+
+  // Voter is true if this server has a vote in the cluster. This might
+  // be false if the server is staging and still coming online.
+  bool Voter = 5;
+}
+
+message RaftConfiguration {
+  repeated RaftServer Servers = 1;
+}
+
+message AutopilotServer {
+  string ID = 1;
+  bool Healthy = 2;
+}
+
+message AutopilotStatus {
+  bool Healthy = 1;
+  repeated AutopilotServer Servers = 2;
+}
+
+message RaftStatus {
+  RaftConfiguration RaftConfiguration = 1;
+  AutopilotStatus AutopilotStatus = 2;
+  repeated string QuorumWarnings = 3;
+}
+
+message GetClusterStatusResponse {
+  string ClusterID = 1;
+  repeated HANode HANodes = 2;
+  bool HAEnabled = 3;
+  RaftStatus RaftStatus = 4;
```

Comment:
I'm not sure what generic storage information we would want to expose. Do you have any examples?

Reply:
I don't really know right now, I have to dig into that :) 
An example info is the node ID I guess. I think both Raft and Consul would have node IDs. 
Would you think that introducing a new message to collect node types would be useful? Something like:
```
message ServerNodes {
RaftStatus RaftStatus = 1;
}
```
This way if we intend to add info for Consul, or any other node with a backend other than Raft would be just added to that message? 
I am just wondering if in the future we would want to extend this, how would it look like. This is particularly important as the proto messages can only be extended and not modified. 
The current format is pretty extendable as well though, so I don't have a strong feeling about my suggestion.

Reply:
That is an interesting point. Having `RaftStatus` sitting at the top might be a bit odd. We could lump it under `Storage`. We have since added a `StorageType`. It might be worthwhile to have a top-level `Storage` message that includes `Type` and then any backend-specific details (`RaftStatus` for now). I will defer to @biazmoreira given that she will be consuming these fields.

Reply:
I'm ok with your suggestion, @ccapurso, no objections. :)